core/src/main/scala/org/apache/spark/input/PortableDataStream.scala

```diff
@@ -194,5 +195,8 @@ class PortableDataStream(
   }
 
   def getPath(): String = path
+
+  @Since("2.2.0")
```

Comment:
This is a public class so I thought adding a `since` tag would benefit the documentation. If it's not desired I can certainly remove it.

As for making the lazy val public vs private: I'm following the style used already in the class. There are public get methods for each private field. I'm not partial to either approach but prefer to keep it consistent.